jax/_src/scipy/linalg.py

```diff
@@ -433,6 +433,11 @@ def triu(m: ArrayLike, k: int = 0) -> Array:
 @_wraps(scipy.linalg.expm, lax_description=_expm_description)
 @partial(jit, static_argnames=('upper_triangular', 'max_squarings'))
 def expm(A: ArrayLike, *, upper_triangular: bool = False, max_squarings: int = 16) -> Array:
+  A, = promote_dtypes_inexact(A)
+
+  if A.ndim > 2:
+    return vmap(partial(expm, upper_triangular=upper_triangular, max_squarings=max_squarings))(A)
```

Comment:
It might be better to do in a loop rather than recursively, but this way it's clearly correct!

Reply:
Switched to `jnp.vectorize` instead, which is a bit more direct.